File: backend/src/main/java/com/maeasoftworks/alfaconverterapi/services/Logger.java
package com.maeasoftworks.alfaconverterapi.services;

import com.maeasoftworks.alfaconverterapi.dao.Log;
import com.maeasoftworks.alfaconverterapi.repository.LogRepository;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
public class Logger {
    private final LogRepository logRepository;

    public Logger(LogRepository logRepository) {
        this.logRepository = logRepository;
    }

    @Transactional
    public void write(Log log) {
        logRepository.save(log);
    }
}

File: backend/src/main/kotlin/com/maeasoftworks/alfaconverter/services/Logger.kt
package com.maeasoftworks.alfaconverter.services

import com.maeasoftworks.alfaconverter.dao.Log
import com.maeasoftworks.alfaconverter.repository.LogRepository
import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Service
class Logger(private val logRepository: LogRepository) {
	@Transactional
	fun write(log: Log) {
		logRepository.save(log)
	}
}